presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/RemoveUnsupportedDynamicFilters.java

```diff
@@ -285,6 +306,39 @@ private Expression removeDynamicFilters(Expression expression, Set<DynamicFilter
                     .collect(toImmutableList()));
         }
 
+        private boolean isSupportedDynamicFilterExpression(Expression expression)
+        {
+            if (expression instanceof SymbolReference) {
+                return true;
+            }
+            if (!(expression instanceof Cast)) {
+                return false;
+            }
+            Cast castExpression = (Cast) expression;
+            if (!(castExpression.getExpression() instanceof SymbolReference)) {
+                return false;
+            }
+            Map<NodeRef<Expression>, Type> expressionTypes = typeAnalyzer.getTypes(session, types, expression);
+            Type probeSymbolType = expressionTypes.get(NodeRef.of(castExpression.getExpression()));
+            Type buildSymbolType = expressionTypes.get(NodeRef.<Expression>of(castExpression));
```

Comment:
it's not build symbol. It's still probe symbol, but after casting. Maybe call them:
```
castSourceType
castTargetType
```